software/ADPL_electron/ADPL_electron.ino

```diff
@@ -124,27 +128,32 @@ void loop() {
 
 
     currentTime = millis();
-    if((currentTime - WAIT_TIME) > ((3600 * VOLUME) *  (1 / OPTIMAL_FLOW)) && (ISITUP == 0)) {
-      pinchValve.up = true;
-      pinchValve.resolution = BATCH_MOVEMENT; // 3mm , make variable
-      WAIT_TIME = currentTime;
-      ISITUP = 1;
-      }
-
-    if((bucket.tip) && (ISITUP == 1)){
-        pinchValve.down = true;
-        pinchValve.resolution = BATCH_MOVEMENT; // 3mm , make variable!
-        bucket.tip = false;
-        ISITUP = 0;
-        WAIT_TIME = millis();
+    if ((currentTime - pinchValve.wait_time) > ((3600 * VOLUME) *  (1 / OPTIMAL_FLOW)) && (pinchValve.down)) {
+        // if the difference between current and wait times > the volume of the bucket ((L*s)/h) *
+        // the inverse of the optimal flow (hrs/L) (units cancel, leaving a number in seconds) AND
+        // the pinch valve is down
+        Log.info("Raising pinch valve...");
+        pinchValve.up = true;
+        pinchValve.resolution = BATCH_MOVEMENT;
+        pinchValve.wait_time = currentTime;
+        Log.info("Pinch valve raised");
     }
 
-    if((bucket.tip) && (ISITUP == 0)){
+    // handle bucket tip
+    if (bucket.tip) {
+        Log.info("Bucket has tipped.");
+        // if bucket is tipped
+        if(pinchValve.up){
+            Log.info("Pinch valve is up. Lowering...");
+            // if the pinch valve is up
+            pinchValve.down = true;
```

Comment:
We don't need both; I have removed `pinchValve.down` and replaced it with `!pinchValve.up` in all locations and wherever `pinchValve.down` was set, `pinchValve.up` is now set to the opposite value